rskj-core/src/main/java/co/rsk/net/sync/PendingMessages.java

```diff
@@ -0,0 +1,46 @@
+package co.rsk.net.sync;
+
+import co.rsk.net.messages.MessageType;
+import co.rsk.net.messages.MessageWithId;
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class PendingMessages {
+
+    private Map<Long, MessageType> messages = new HashMap<>();
+    private long lastRequestId;
+
+    public void register(MessageWithId message) {
+        this.messages.put(message.getId(), message.getResponseMessageType());
+    }
+
+    public long getNextRequestId(){
+        return ++lastRequestId;
+    }
+
+    public boolean isPending(long responseId, MessageType type) {
```

Comment:
This should take a MessageWithId probably. The same way as register takes one.

Reply:
Yes, I improved register and forgot this one. It will improve readibility.